velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -63,28 +73,52 @@ SubstraitVeloxExprConverter::toVeloxExpr(
   auto typeCase = sLit.literal_type_case();
   switch (typeCase) {
     case ::substrait::Expression_Literal::LiteralTypeCase::kFp64:
-      return std::make_shared<core::ConstantTypedExpr>(sLit.fp64());
+      return std::make_shared<core::ConstantTypedExpr>(
+          velox::variant(velox::variant(sLit.fp64())));
     case ::substrait::Expression_Literal::LiteralTypeCase::kBoolean:
-      return std::make_shared<core::ConstantTypedExpr>(sLit.boolean());
+      return std::make_shared<core::ConstantTypedExpr>(variant(sLit.boolean()));
+    case ::substrait::Expression_Literal::LiteralTypeCase::kI64:
+      return std::make_shared<core::ConstantTypedExpr>(variant(sLit.i64()));
     default:
       VELOX_NYI(
           "Substrait conversion not supported for type case '{}'", typeCase);
   }
 }
 
+std::shared_ptr<const core::ITypedExpr>
+SubstraitVeloxExprConverter::toVeloxExpr(
+    const ::substrait::Expression::Cast& sCast,
+    const RowTypePtr& vPreNodeOutPut) {
+  auto subType = subParser_->parseType(sCast.type());
+  auto vCastType = toVeloxType(subType->type);
+  // TODO add flag in substrait after. now is set false.
```

Comment:
Perhaps, file an issue with Substrait and reference it here?

Reply:
See https://github.com/substrait-io/substrait/issues/185

Reply:
Thanks for filing an issue. Would you add a link to the comment in the code?